check.py

```diff
@@ -93,43 +121,44 @@ def setup_waterfall():
   except Exception, e:
     warn('could not run vanilla LLVM from waterfall: ' + str(e) + ', looked for clang at ' + CLANG)
 
-fetch_waterfall()
-setup_waterfall()
+if options.test_waterfall:
+  fetch_waterfall()
+  setup_waterfall()
 
-if only_prepare:
+if options.only_prepare:
   print 'waterfall is fetched and setup, exiting since --only-prepare'
   sys.exit(0)
 
 # external tools
 
 has_node = False
 try:
-  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  subprocess.check_call(['nodejs', '--version'], stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
```

Comment:
I did some experiments on Linux. Apparently `shell=True` has some effect on the stdin used by the process; if you feed something in, e.g. 'stdin=open('/dev/null')` it works. Maybe nodejs is trying to read something from stdin?. `shell=True` also affects stdout; i.e. if you set your command to something like `['echo','foo']` the output will appear on your console with `shell=False` but not with `shell=True`. In other words, stdin/out/err are supposed to be inherited by the child process when not using redirection, but apparently `shell=True` has some other behavior (that doesn't seem to be documented as far as I can tell).

Anyway it's odd and magical and makes me want to avoid `shell=True` even more :-/. Unfortunately although Python has lots of good shell-like tools, I don't think an equivalent to `which`/`where` is among them. However it's fairly straightforward to write: see e.g. https://chromium.googlesource.com/native_client/src/native_client/+/master/pynacl/file_tools.py#72